contract/src/main/kotlin/io/provenance/scope/loan/contracts/RecordStandaloneLoanValidationResultsContract.kt

```diff
@@ -0,0 +1,41 @@
+package io.provenance.scope.loan.contracts
+
+import io.provenance.scope.contract.annotations.Function
+import io.provenance.scope.contract.annotations.Input
+import io.provenance.scope.contract.annotations.Participants
+import io.provenance.scope.contract.annotations.Record
+import io.provenance.scope.contract.annotations.ScopeSpecification
+import io.provenance.scope.contract.proto.Specifications.PartyType
+import io.provenance.scope.contract.spec.P8eContract
+import io.provenance.scope.loan.LoanScopeFacts
+import io.provenance.scope.loan.LoanScopeInputs
+import io.provenance.scope.loan.utility.ContractRequirementType
+import io.provenance.scope.loan.utility.loanValidationResultsValidation
+import io.provenance.scope.loan.utility.validateRequirements
+import tech.figure.validation.v1beta2.LoanValidation
+import tech.figure.validation.v1beta2.ValidationIteration
+import tech.figure.validation.v1beta2.ValidationResponse
+
+@Participants(roles = [PartyType.VALIDATOR])
+@ScopeSpecification(["tech.figure.loan"])
+open class RecordStandaloneLoanValidationResultsContract(
+    @Record(name = LoanScopeFacts.loanValidationMetadata, optional = true) val validationRecord: LoanValidation?,
+) : P8eContract() {
+
+    @Function(invokedBy = PartyType.VALIDATOR)
+    @Record(LoanScopeFacts.loanValidationMetadata)
+    open fun recordStandaloneLoanValidationResults(
+        @Input(LoanScopeInputs.validationResponse) submission: ValidationResponse
+    ): LoanValidation {
+        validateRequirements(ContractRequirementType.VALID_INPUT) {
+            loanValidationResultsValidation(submission.results)
+        }
+        return (validationRecord?.toBuilder() ?: LoanValidation.newBuilder()).also { recordBuilder ->
+            recordBuilder.addIteration(
+                ValidationIteration.newBuilder().also { iterationBuilder ->
+                    iterationBuilder.results = submission.results
+                }.build()
```

Comment:
Food for thought: do you think we should make the invoker pass a ValidationRequest as well? That way we could at least fill in a description, requester/validator to give more context to people auditing this data later.

Reply:
I suppose so. Would we then also want to add a field / set a value somewhere to indicate that the request & response were set by the same party?